pkg/apis/core/validation/validation.go

```diff
@@ -5051,8 +5061,20 @@ func ValidateConfigMap(cfg *core.ConfigMap) field.ErrorList {
 func ValidateConfigMapUpdate(newCfg, oldCfg *core.ConfigMap) field.ErrorList {
 	allErrs := field.ErrorList{}
 	allErrs = append(allErrs, ValidateObjectMetaUpdate(&newCfg.ObjectMeta, &oldCfg.ObjectMeta, field.NewPath("metadata"))...)
-	allErrs = append(allErrs, ValidateConfigMap(newCfg)...)
 
+	if oldCfg.Immutable != nil && *oldCfg.Immutable {
```

Comment:
Does this check also need to be behind feature gate?

Reply:
My understanding is it doesn't have to.
The rationale is:
- if the feature gate is off, then the "dropDisabledFields" logic from PrepareForUpdate is firing
- this is happening before validation, so the Immutable field will be clear in this case

Reply:
SGTM